File: P_2/Napakalaki_Java_P2/src/napakalaki/Dice.java
// MADE BY: JOSE MIGUEL HDEZ GARCIA & JAIRO ABRIL MOYA

package napakalaki;
import java.util.Random;
// Clase singleton

/**
 *
 * @author ixjosemi
 */

public class Dice {

    private static Dice instance = null;

    // Dado que el constructor es privado no puede instanciarse desde otra clase
    private Dice(){

    }

    // Consultor que devuelve instance
    public static Dice getInstance(){
        return instance;
    }

    // Generador de numeros aleatorios entre 1 y 6 (Las caras del dado)
    public int nextNumber(){
        Random rnd = new Random();
        int random = rnd.nextInt() * 6 + 1;

        return random;
    }

}

File: P_2/Examen_P2/Napakalaki_Ruby_P2/lib/dice.rb
#encoding: utf-8
# MADE BY: JOSE MIGUEL HDEZ GARCIA & JAIRO ABRIL MOYA

module Napakalaki
require "singleton"

class Dice
    
    # Hacemos uso del singleton
    include Singleton
    
    def next_number
        return 1 + rand(6)
    end
end
end
